src/main/java/com/google/sps/managers/DatastoreManager.java

```diff
@@ -0,0 +1,334 @@
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+/** Servlet responsible for implementing the functions in the interface. **/
+public class DatastoreManager implements IDataManager {
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  Gson gson = new Gson();
+
+  /**
+  * Creates or Replaces an entity of a single User data in datastore
+  * @param user a User variable representing a single instance of a user.
+  */
+  public void createOrReplaceUser(User user) {
+    Entity userEntity = new Entity("User", user.getUserID());
+    userEntity.setProperty("username", user.getUsername());
+    userEntity.setProperty("firstname", user.getFirstName());
+    userEntity.setProperty("lastname", user.getLastName());
+    userEntity.setProperty("gamesCreated", user.getGamesCreated());
+    userEntity.setProperty("profilePictureUrl", user.getProfilePictureUrl());
+    String numbersJson = gson.toJson(user.getGamesCompletedWithTime());
+    userEntity.setProperty("gamesCompletedWithTime", numbersJson);
+    datastore.put(userEntity);
+  }
+  
+  /**
+    * Retrieves a single user entity from the datastore.
+    * @param userID the unique code used to identify this specific user.
+    * @return a user object with the properties specified in the Builder.
+  */
+  public User retrieveUser(String userID) {
+    Key userEntityKey = KeyFactory.createKey("User", userID);
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(userEntityKey);
+    }
+    catch(Exception e) {
+      return null;
+    }
+    String userName = (String) userEntity.getProperty("username");
+    String firstName = (String) userEntity.getProperty("firstname");
+    String lastName = (String) userEntity.getProperty("lastname");
+    ArrayList<String> gamesCreated = (ArrayList<String>) userEntity.getProperty("gamesCreated");
+    String profilePictureUrl = (String) userEntity.getProperty("profilePictureUrl");
+    String gamesCompletedWithTimeJson = (String) userEntity.getProperty("gamesCompletedWithTime");
+    Type pairListType = new TypeToken<ArrayList<Pair<String, Long>>>(){}.getType();
+    ArrayList<Pair<String, Long>> gamesCompletedWithTime = gson.fromJson(gamesCompletedWithTimeJson, pairListType);
+  
+    User.Builder user = new User.Builder(userID).setUsername(userName).setFirstName(firstName).setLastName(lastName);
+    user.setProfilePictureUrl(profilePictureUrl).setGamesCreated(gamesCreated);
+    user.setGamesCompletedWithTime(gamesCompletedWithTime);
+    return user.build();
+  }
+
+  /**
+    * Creates or Replaces the static data of a single game in datastore as an entity
+    * @param game an Game variable representing a single instance of a game.
```

Comment:
Keep the spacing consistent so have 
/*
*
*
*/

rather than 
/*
  *
  *
*/

Here and other places